### PR TITLE
Added support for authentication, which is required since Neo4J 2.2

### PR DESCRIPTION
**Pull request by pvoosten**
I made the changes to be able to work with the newest Neo4J release.

### CODE REVIEW

File: src/CypherNet.Core/GraphStore.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class GraphStore
         #region Constructors and Destructors
 
         /// <summary>
-        /// Initialises a new instance of the <see cref="GraphStore"/> class.
+        /// Initialises a new instance of the <see cref="GraphStore"/> class, usable before Neo4J version 2.2.0.
```
Comment (mtranter) [inline src/CypherNet.Core/GraphStore.cs]: One very minor comment on this.  This constructor will still be usable with 2.2.0, but the server will have to be configured to not use auth: http://neo4j.com/docs/stable/security-server.html#security-server-auth

Comment (pvoosten) [inline src/CypherNet.Core/GraphStore.cs]: I agree, the comment for the constructor is incorrect.

I'm creating an application with CypherNet.Core in an STA thread. When I get that decently working, I will add unit tests and push the changed comment. For the moment, the application hangs when executing GraphStore.Initialize.

**Comment by pvoosten** [comment]
Can a LICENSE file be added? I don't care whether it's open source or GPL, but currently, I can't use CypherNet.Core in real world projects..

**Comment by mtranter** [comment]
@mikehancock  Any thoughts mate?

**Pull request merged**